### DISCUSSION TITLE
What is the reason to only support maven-processor-plugin for APT when using maven?

### DISCUSSION THREAD

**Discussion by mpet** [eclipse-m2e/m2e-core#660] [General]
Hi,
In the gui for Maven Annoation Processing there is only one option for maven-processor-plugin.
![image](https://user-images.githubusercontent.com/1445826/165743768-e2d78c39-a4b8-476a-965d-e510de33945c.png)

I checked this plugin and found there are classes for maven-compiler-plugin.

![image](https://user-images.githubusercontent.com/1445826/165743916-de66912d-1ea0-4cda-8dbe-9187069502d0.png)

Anyone knows why? @fbricon  do you?

//mike

**Comment by fbricon** [comment]
see https://developer.jboss.org/en/tools/blog/2012/05/20/annotation-processing-support-in-m2e-or-m2e-apt-100-is-out

**Reply by mickaelistria** [reply]
"The downside of using JDT APT is, there's no separation between main and test classes (the way maven-processor-plugin handles them). " is not really true anymore.
@mpet if you're willing to contribute suppot for JDT's APT in m2e, it would be very welcome!
